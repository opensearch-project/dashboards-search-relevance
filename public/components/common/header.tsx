/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { EuiText, EuiLink, EuiPanel, EuiTitle } from '@elastic/eui';

interface HeaderProps {
  children?: React.ReactNode;
}

export const Header = ({ children }: HeaderProps) => {
  return (
    <EuiPanel
      hasBorder={false}
      hasShadow={false}
      grow={false}
      borderRadius="none"
      style={{ borderBottom: '1px solid #D3DAE6' }}
    >
      <EuiTitle>
        <h1>Compare search results</h1>
      </EuiTitle>
      <EuiText>
        <p>
          Compare results using the same search text with different queries.{' '}
          <EuiLink
            href="https://opensearch.org/docs/latest/search-plugins/search-relevance"
            target="_blank"
          >
            Learn more
          </EuiLink>
        </p>
      </EuiText>
      {children}
    </EuiPanel>
  );
};
