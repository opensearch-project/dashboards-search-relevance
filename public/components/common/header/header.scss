/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

.search-relevance {
  .header {
    padding: 12px 20px;
    background-color: #FFFFFF;
    border-bottom: 1px solid #D3DAE6;
  }
}
