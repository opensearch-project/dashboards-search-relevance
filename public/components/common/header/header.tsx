/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { EuiPageHeader, EuiCallOut, EuiText, EuiLink } from '@elastic/eui';

import './header.scss';

export const Header = () => {
  return (
    <EuiPageHeader pageTitle="Compare Search results" className="header">
      <EuiCallOut title="Experimental Feature" iconType="iInCircle">
        <EuiText>
          <p>
            The feature is experimental. For more information, see{' '}
            <EuiLink>Compare Search Results Documentation.</EuiLink> To leave feedback, visit{' '}
            <EuiLink>forums.opensearch.com.</EuiLink>
          </p>
        </EuiText>
      </EuiCallOut>
    </EuiPageHeader>
  );
};
