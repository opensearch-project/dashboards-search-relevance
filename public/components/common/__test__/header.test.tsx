/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { waitFor } from '@testing-library/react';
import { Header } from '../header';

describe('Header component', () => {
  configure({ adapter: new Adapter() });

  it('Renders header component', async () => {
    const wrapper = mount(<Header />);

    wrapper.update();

    await waitFor(() => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
