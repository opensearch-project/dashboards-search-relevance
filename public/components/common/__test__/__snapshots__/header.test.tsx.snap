// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Renders header component 1`] = `
<Header>
  <EuiPanel
    borderRadius="none"
    color="transparent"
    grow={false}
    hasBorder={false}
    hasShadow={false}
  >
    <div
      className="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusNone euiPanel--transparent euiPanel--noShadow euiPanel--noBorder euiPanel--flexGrowZero"
    >
      <EuiTitle>
        <h1
          className="euiTitle euiTitle--medium"
        >
          Compare search results
        </h1>
      </EuiTitle>
      <EuiText>
        <div
          className="euiText euiText--medium"
        >
          <p>
            Compare results using the same search text with different queries.
             
            <EuiLink
              href="https://opensearch.org/docs/latest/search-plugins/search-relevance"
              target="_blank"
            >
              <a
                className="euiLink euiLink--primary"
                href="https://opensearch.org/docs/latest/search-plugins/search-relevance"
                rel="noopener noreferrer"
                target="_blank"
              >
                Learn more
                <EuiIcon
                  aria-label="External link"
                  className="euiLink__externalIcon"
                  size="s"
                  type="popout"
                >
                  <EuiIconEmpty
                    aria-hidden={true}
                    aria-label="External link"
                    className="euiIcon euiIcon--small euiIcon-isLoading euiLink__externalIcon"
                    focusable="false"
                    role="img"
                    style={null}
                  >
                    <svg
                      aria-hidden={true}
                      aria-label="External link"
                      className="euiIcon euiIcon--small euiIcon-isLoading euiLink__externalIcon"
                      focusable="false"
                      height={16}
                      role="img"
                      style={null}
                      viewBox="0 0 16 16"
                      width={16}
                      xmlns="http://www.w3.org/2000/svg"
                    />
                  </EuiIconEmpty>
                </EuiIcon>
                <EuiScreenReaderOnly>
                  <span
                    className="euiScreenReaderOnly"
                  >
                    <EuiI18n
                      default="(opens in a new tab or window)"
                      token="euiLink.newTarget.screenReaderOnlyText"
                    >
                      (opens in a new tab or window)
                    </EuiI18n>
                  </span>
                </EuiScreenReaderOnly>
              </a>
            </EuiLink>
          </p>
        </div>
      </EuiText>
    </div>
  </EuiPanel>
</Header>
`;
